lib/mariaex/row_parser.ex

```diff
@@ -0,0 +1,216 @@
+    << 0 :: 8, null_bitfield :: size(nullbin_size)-little-unit(8), rest :: binary >> = packet
+    decode_bin_rows(rest, fields, null_bitfield >>> 2, [])
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [_ | fields], nullint, acc) when (nullint &&& 1) === 1 do
+    decode_bin_rows(rest, fields, nullint >>> 1, [nil | acc])
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:string | fields], null_bitfield,  acc) do
+    decode_string(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:uint8 | fields], null_bitfield, acc) do
+    decode_uint8(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:int8 | fields], null_bitfield, acc) do
+    decode_int8(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:uint16 | fields], null_bitfield, acc) do
+    decode_uint16(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:int16 | fields], null_bitfield, acc) do
+    decode_int16(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:uint32 | fields], null_bitfield, acc) do
+    decode_uint32(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:int32 | fields], null_bitfield, acc) do
+    decode_int32(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:uint64 | fields], null_bitfield, acc) do
+    decode_uint64(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:int64 | fields], null_bitfield, acc) do
+    decode_int64(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:year | fields], null_bitfield, acc) do
+    decode_int16(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:time | fields], null_bitfield, acc) do
+    decode_time(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:date | fields], null_bitfield, acc) do
+    decode_date(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:datetime | fields], null_bitfield, acc) do
+    decode_datetime(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:decimal | fields], null_bitfield, acc) do
+    decode_decimal(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:float32 | fields], null_bitfield, acc) do
+    decode_float32(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:float64 | fields], null_bitfield, acc) do
+    decode_float64(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:bit | fields], null_bitfield, acc) do
+    decode_string(rest, fields, null_bitfield >>> 1, acc)
+  end
+
+  defp decode_bin_rows(<<rest::bits>>, [:nil | fields], null_bitfield, acc) do
```

Comment:
Is this every reachable given the null bitfield? Previously no such clause but is ok for completeness.